pkg/query-service/model/v3/v3.go

```diff
@@ -0,0 +1,240 @@
+	AggregateOperatorMin           AggregateOperator = "min"
+	AggregateOperatorMax           AggregateOperator = "max"
+	AggregateOperatorP05           AggregateOperator = "p05"
+	AggregateOperatorP10           AggregateOperator = "p10"
+	AggregateOperatorP20           AggregateOperator = "p20"
+	AggregateOperatorP25           AggregateOperator = "p25"
+	AggregateOperatorP50           AggregateOperator = "p50"
+	AggregateOperatorP75           AggregateOperator = "p75"
+	AggregateOperatorP90           AggregateOperator = "p90"
+	AggregateOperatorP95           AggregateOperator = "p95"
+	AggregateOperatorP99           AggregateOperator = "p99"
+	AggregateOperatorRate          AggregateOperator = "rate"
+	AggregateOperatorSumRate       AggregateOperator = "sum_rate"
+	AggregateOperatorAvgRate       AggregateOperator = "avg_rate"
+	AggregateOperatorMinRate       AggregateOperator = "min_rate"
+	AggregateOperatorMaxRate       AggregateOperator = "max_rate"
+	AggregateOperatorRateSum       AggregateOperator = "rate_sum"
+	AggregateOperatorRateAvg       AggregateOperator = "rate_avg"
+	AggregateOperatorRateMin       AggregateOperator = "rate_min"
+	AggregateOperatorRateMax       AggregateOperator = "rate_max"
+	AggregateOperatorHistQuant50   AggregateOperator = "hist_quantile_50"
+	AggregateOperatorHistQuant75   AggregateOperator = "hist_quantile_75"
+	AggregateOperatorHistQuant90   AggregateOperator = "hist_quantile_90"
+	AggregateOperatorHistQuant95   AggregateOperator = "hist_quantile_95"
+	AggregateOperatorHistQuant99   AggregateOperator = "hist_quantile_99"
+)
+
+type ReduceToOperator string
+
+const (
+	ReduceToOperatorLast ReduceToOperator = "last"
+	ReduceToOperatorSum  ReduceToOperator = "sum"
+	ReduceToOperatorAvg  ReduceToOperator = "avg"
+	ReduceToOperatorMin  ReduceToOperator = "min"
+	ReduceToOperatorMax  ReduceToOperator = "max"
+)
+
+type QueryType string
+
+const (
+	QueryTypeBuilder       QueryType = "builder"
+	QueryTypeClickHouseSQL QueryType = "clickhouse_sql"
+	QueryTypePromQL        QueryType = "promql"
+)
+
+type PanelType string
+
+const (
+	PanelTypeValue PanelType = "value"
+	PanelTypeGraph PanelType = "graph"
+	PanelTypeTable PanelType = "table"
+	PanelTypeList  PanelType = "list"
+)
+
+// AggregateAttributeRequest is a request to fetch possible attribute keys
+// for a selected aggregate operator and search text.
+// The context of the selected aggregate operator is used as the
+// type of the attribute key is different for different aggregate operators.
+// For example, for the aggregate operator "avg" the attribute value type must be
+// a number
+type AggregateAttributeRequest struct {
+	DataSource DataSource        `json:"dataSource"`
+	Operator   AggregateOperator `json:"aggregateOperator"`
+	SearchText string            `json:"searchText"`
+	Limit      int               `json:"limit"`
+}
+
+type TagType string
+
+const (
+	TagTypeColumn   TagType = "column"
+	TagTypeTag      TagType = "tag"
```

Comment:
Are we using `tag` and `attributes` interchangeably

Reply:
Yes